File: packman/src/main/java/fkn/dlaskina/packman/map/TimerTaskRedraw.java
package fkn.dlaskina.packman.map;

import java.awt.Graphics;
import java.util.TimerTask;

import fkn.dlaskina.packman.panels.MapPanel;
import org.apache.log4j.LogManager;
import org.apache.log4j.Logger;

/**
 * Этот класс отвечает за движение.
 * @author Vladimir Laskin
 * @version 1.0
 */
public class TimerTaskRedraw extends TimerTask {

    private static final Logger LOG = LogManager.getLogger(TimerTaskRedraw.class);

    private MapPanel mapPanel;
    private int frame = 0;

    public TimerTaskRedraw(final MapPanel mapPanel) {
        this.mapPanel = mapPanel;
    }

    @Override
    public void run() {
        final Graphics gr = mapPanel.getGraphics();
        if (gr != null) {
            mapPanel.setRedrawMap(true);
            mapPanel.setFrame(frame);
            if (++frame > 39) frame = 0;
            mapPanel.update(gr);
        }
    }
}

File: packman/src/main/java/fkn/dlaskina/packman/timers/TimerTaskRedraw.kt
package fkn.dlaskina.packman.timers

import fkn.dlaskina.packman.panels.MapPanel
import mu.KotlinLogging
import java.util.*

private val log = KotlinLogging.logger{}

/**
 * Этот класс отвечает за движение.
 * @author Vladimir Laskin
 * @version 1.0
 */
class TimerTaskRedraw(private val mapPanel: MapPanel) : TimerTask() {

    private var frame = 0
    override fun run() {
        val gr = mapPanel.graphics
        if (gr != null) {
            mapPanel.setRedrawMap(true)
            mapPanel.setFrame(frame)
            if (++frame > 39) frame = 0
            mapPanel.update(gr)
        }
    }

}